### ISSUE TITLE
Speed difference between window app and console app

### ISSUE THREAD

**Issue by hoangvh94** [bupticybee/TexasSolver#170]
A big difference about speed between the window app and console app

![image](https://github.com/bupticybee/TexasSolver/assets/45188220/624da142-7607-461b-9092-301b34ee139f)

and consolve_solver.exe (left is wndow app, right is console app built)
![image](https://github.com/bupticybee/TexasSolver/assets/45188220/cbb6c88e-e0d0-4433-bc20-4b83d98b97ec)

**Comment by bupticybee** [comment]
<img width="1037" alt="截屏2023-06-12 10 32 43" src="https://github.com/bupticybee/TexasSolver/assets/9393610/c1f9ecdb-f8eb-483b-bde0-674c9a812fd9">
![截屏2023-06-12 10 32 52](https://github.com/bupticybee/TexasSolver/assets/9393610/25e854f3-3f97-4a7f-9f25-a2b708844604)


I done some experiments, yes,  it seems that in small tree settings, the two does run at a different speed. Since both binary is compiled by the same core code. I would guess it's the compiler that matters.

BTW, it's more common in my experiments if the tree are big enough, console version is faster.

**Comment by hoangvh94** [comment]
why its (console_solver.exe) so different between Window app (3807 kb) and Console built (657 kb)

and when I copied and ran console_solver(3807) 
![image](https://github.com/bupticybee/TexasSolver/assets/45188220/7e0ec802-0522-42ec-9ab4-6c7f190e5974)
instead of (657) one 
![image](https://github.com/bupticybee/TexasSolver/assets/45188220/b0305f59-5387-4b1e-978f-81d65dee216c)

it's extremely fast 
![image](https://github.com/bupticybee/TexasSolver/assets/45188220/f5c7888d-45d5-434a-82c0-d5d99a7b8227)
![image](https://github.com/bupticybee/TexasSolver/assets/45188220/9bf77dbc-aad8-4587-87f9-b55550cf71bd)
and the process log is a bit different in picture 1st comment above
![image](https://github.com/bupticybee/TexasSolver/assets/45188220/53ed28da-fd2d-4e4c-be79-160edcb90d3e)

**Comment by bupticybee** [comment]
> 

They are compiled under different environment, different compiler, and some different code. Size different and speed different are excepted.

**Comment by bupticybee** [comment]
> 

If you are that into speed, I encourage you to build the binary youself to get best speed.

**Comment by hoangvh94** [comment]
you are right, I tried some others environment and speed is different. So could you tell me which environment I should use for faster speed? require for RAM or CPU ....?

**Comment by bupticybee** [comment]
> you are right, I tried some others environment and speed is different. So could you tell me which environment I should use for faster speed? require for RAM or CPU ....?

I really don't know about that, I think this kind of knowledge requires a lot of hardware and experiments, I don't have both.
